packages/components/src/Popover/Popover.tsx

```diff
@@ -55,14 +57,48 @@ export interface PopoverProps extends UsePopoverProps {
    * The element which hovering on/off of will show/hide the triggering element
    */
   hoverDisclosureRef?: HTMLElement | null | RefObject<HTMLElement>
+  /**
+   * Convert popover to full-screen dialog on mobile
+   * @default false
+   */
+  renderMobileDialog?: boolean
+  /**
+   * Populate dialog header content when renderMobileDialog is true
+   */
+  dialogHeader?: string
 }
 
 export const Popover = ({
   children,
   hoverDisclosureRef,
+  renderMobileDialog = false,
+  dialogHeader,
+  content,
   ...props
 }: PopoverProps) => {
-  const { domProps, isOpen, popover } = usePopover(props)
+  const { domProps, isOpen, popover } = usePopover({
+    ...props,
+    content,
+  })
+  const dialogProps = useDialog({
+    content: (
+      <>
+        <DialogHeader
+          borderBottom="1px solid"
+          borderBottomColor="ui2"
+          position="sticky"
+          top="0"
+          p="medium"
+          px="medium"
+          pr="medium"
+        >
+          {dialogHeader}
+        </DialogHeader>
+        {content}
```

Comment:
I'm pretty retiscent to open up lots of new behavior on `DialogHeader` and I think a simple recomposition will allow you to unwind all of the changes to `Dialog*`

```suggestion
  <DialogHeader>{dialogHeader}</DialogHeader>
  <DialogContent>{content}</DialogContent>
```

Reply:
I can't really use `DialogContent` here though because in this context the `content` is likely already wrapped in `PopoverContent`, so it would be doubling up the whitespace. 

Also, jared specified smaller margins in mobile, and it's REALLY helpful to have the header be sticky on iPhone SE:

![unnamed](https://user-images.githubusercontent.com/238293/101960287-f88f5c00-3bbb-11eb-9842-d22a2fb747d4.png)

Maybe DialogHeader should just be responsive to begin with?

Reply:
Ah ok, I'm following now. So the content would scroll rather than the surface. That makes sense. Maybe I can just strip the padding from `DialogContent`, and then make the header more responsive